rclcpp_action/test/test_client.cpp

```diff
@@ -326,11 +329,34 @@ TEST_F(TestClient, construction_and_destruction_callback_group)
     ).reset());
 }
 
+TEST_F(TestClient, construction_and_destruction_rcl_errors)
+{
+  {
+    auto mock = mocking_utils::patch_and_return(
+      "lib:rclcpp_action", rcl_action_client_fini, RCL_RET_ERROR);
+    // It just logs an error message and continues
+    EXPECT_NO_THROW(
+      rclcpp_action::create_client<ActionType>(client_node, action_name).reset());
+  }
+  {
+    auto mock = mocking_utils::patch_and_return(
+      "lib:rclcpp_action", rcl_action_client_init, RCL_RET_ERROR);
+    EXPECT_THROW(
+      rclcpp_action::create_client<ActionType>(client_node, action_name).reset(),
+      rclcpp::exceptions::RCLError);
+  }
+  {
+    auto mock = mocking_utils::patch_and_return(
+      "lib:rclcpp_action", rcl_action_client_wait_set_get_num_entities, RCL_RET_ERROR);
+    EXPECT_THROW(
+      rclcpp_action::create_client<ActionType>(client_node, action_name),
+      rclcpp::exceptions::RCLError);
+  }
+}
+
 TEST_F(TestClient, wait_for_action_server)
 {
   auto action_client = rclcpp_action::create_client<ActionType>(client_node, action_name);
-  EXPECT_FALSE(action_client->wait_for_action_server(0ms));
```

Comment:
These checks were flaky on my machine

Reply:
Really? The server's not up yet, how could it return `true`. Perhaps an issue with the ROS graph?

Reply:
It must have been a weird issue. Also it turns out these two checks were responsible for about 20 lines of coverage. I added them back in.